bin/execution.ml

```diff
@@ -84,13 +140,25 @@ let rec evaluate_one_step mode expression_list stack_ =
         | _, _ -> stack := Stack.push (String { value = "()" }) stack'')
     (*do the rest % (modulo) *)
     | _ -> failwith "Invalid operator"
+  and construct_float_number mode token =
+    let operand, stack' = Stack.pop !stack in
+    match operand with
+    | Float f ->
+        let new_value =
+          f.value
+          +. float_of_int (int_of_char token - int_of_char '0')
```

Comment:
when you want, you can put this `(int_of_char token - int_of_char '0')` in an function outside, so that the code becomes clearer (also use the function in the other places)